concurrency-limits-core/src/test/java/com/netflix/concurrency/limits/limiter/AbstractPartitionedLimiterTest.java

```diff
@@ -227,4 +236,86 @@ public void testBypassSimpleLimiter() {
+                .partition("A", 0.5)
+                .partition("B", 0.3)
+                .partition("C", 0.2)
+                .build();
+
+        ExecutorService executor = Executors.newFixedThreadPool(THREAD_COUNT * 3);
+        CountDownLatch startLatch = new CountDownLatch(1);
+        CountDownLatch endLatch = new CountDownLatch(THREAD_COUNT * 3);
+        Map<String, AtomicInteger> successCounts = new ConcurrentHashMap<>();
+        Map<String, AtomicInteger> rejectionCounts = new ConcurrentHashMap<>();
+        Map<String, AtomicInteger> maxConcurrents = new ConcurrentHashMap<>();
+        AtomicInteger globalMaxInflight = new AtomicInteger(0);
+
+        for (String partition : Arrays.asList("A", "B", "C")) {
+            successCounts.put(partition, new AtomicInteger(0));
+            rejectionCounts.put(partition, new AtomicInteger(0));
+            maxConcurrents.put(partition, new AtomicInteger(0));
+
+            for (int i = 0; i < THREAD_COUNT; i++) {
+                executor.submit(() -> {
+                    try {
+                        startLatch.await();
+                        for (int j = 0; j < ITERATIONS; j++) {
+                            Optional<Listener> listener = limiter.acquire(partition);
+                            if (listener.isPresent()) {
+                                try {
+                                    int current = limiter.getPartition(partition).getInflight();
+                                    maxConcurrents.get(partition).updateAndGet(max -> Math.max(max, current));
+                                    successCounts.get(partition).incrementAndGet();
+                                    globalMaxInflight.updateAndGet(max -> Math.max(max, limiter.getInflight()));
+                                    Thread.sleep(1); // Simulate some work
+                                } finally {
+                                    listener.get().onSuccess();
+                                }
+                            } else {
+                                rejectionCounts.get(partition).incrementAndGet();
+                            }
+                        }
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                    } finally {
+                        endLatch.countDown();
+                    }
+                });
+            }
+        }
+
+        startLatch.countDown();
+        endLatch.await();
+        executor.shutdown();
+        executor.awaitTermination(10, TimeUnit.SECONDS);
+
+        for (String partition : Arrays.asList("A", "B", "C")) {
+            System.out.println(partition + " success count: " + successCounts.get(partition).get());
+            System.out.println(partition + " rejection count: " + rejectionCounts.get(partition).get());
+            System.out.println(partition + " max concurrent: " + maxConcurrents.get(partition).get());
```

Comment:
nit: can we avoid System.out.println's in the tests? I think it's ok to build an append this string to the assertions if they fail as a hint.

Reply:
fixed in [09a1e6f](https://github.com/Netflix/concurrency-limits/pull/204/commits/09a1e6fd1d649b881c067d18f294011f2360cbe5)